test/test_autograd.py

```diff
@@ -7561,6 +7585,66 @@ def test_non_differentiable(self):
             # No differentiable outputs, shouldn't error
             eq = foo == bar
 
+    def test_create_new_zeros_with_same_meta(self):
+        new_zeroes_fn = torch.ops.aten._new_zeros_with_same_feature_meta
+
+        def check(a, b):
+            def assert_same_meta(t, target):
+                import operator
+
+                for num_bdim in range(t.dim()):
+                    result = new_zeroes_fn(t, target, num_bdim)
+
+                    self.assertEqual(result.dim(), target.dim() + num_bdim)
+
+                    # Check size/strides match for feature dims only
+                    for i in range(num_bdim, result.dim()):
+                        self.assertEqual(result.size()[i], target.size()[i - num_bdim])
+                        self.assertEqual(result.stride()[i], target.stride()[i - num_bdim])
+
+                    # Check that we generate strides reasonably
+                    if target.is_contiguous():
+                        self.assertTrue(result.is_contiguous())
+
+                    self.assertEqual(result.storage_offset(), target.storage_offset())
+
+                    prod_of_t_bdims = reduce(operator.mul, t.size()[:num_bdim], 1)
+                    self.assertEqual(len(result.storage()), len(target.storage()) * prod_of_t_bdims)
+
+                    # TensorOptions is same
+                    self.assertEqual(result.dtype, target.dtype)
+
+            assert_same_meta(a, b)
+            assert_same_meta(b, a)
+
+        a = torch.randn(5, dtype=torch.float)
+        b = torch.randn(2, 3, 4, dtype=torch.double)
```

Comment:
Do we allow `_new_zeros_with_same_feature_meta(self: Tensor[5], other: Tensor[2, 3, 4])`? Because that's what it looks like is happening here.

Reply:
Yup, this is to illustrate the in-place over view case where a view takes `other` (the base) to `self` (which would be the grad). Later in `_set_fw_grad` we'd have to replay the view on this zero tensor before copying the grad (which is self here) into it.